File: code/306/solution.go
package main /*{{{*/

import (
	"fmt"
)

func solve(upper int) int {
	ret := upper / 34 * 5
	specialCase := []int{1, 15, 35}
	for _, i := range specialCase {
		if upper >= i {
			ret++
		}
	}

	normalCase := []int{5, 9, 21, 25, 29}
	for _, i := range normalCase {
		if upper%34 >= i {
			ret++
		}
	}

	return upper - ret
}

func main() {
	fmt.Println(solve(5))
	fmt.Println(solve(50))
	fmt.Println(solve(1000 * 1000))
}

File: code/306/solution.cpp
#include <iostream>/*{{{*/
const long long e3 = 1000;
const long long e6 = e3 * e3;
using namespace std; /*}}}*/
int solve(int upper) {
    int ret    = upper / 34 * 5 + (upper >= 15) + (upper >= 35) + (upper >= 1);
    int data[] = {5, 9, 21, 25, 29};
    for (int i = 0; i < 5; i++)
        if (upper % 34 >= data[i])
            ret++;
    return upper - ret;
}
int main() {
    cout << solve(5) << endl;
    cout << solve(50) << endl;
    cout << solve(e6) << endl;
    return 0;
}
